content/knowledge-base/guides/troubleshoot.md

```diff
@@ -105,6 +105,202 @@ spec:
 > Note that a reference to a `ControllerConfig` can be added to an already
 > installed `Provider` and it will update its `Deployment` accordingly.
 
+## Compositions and XRDs
+
+### General troubleshooting steps
+
+Most error messages are logged to resources' event fields. Whenever your Composite Resources are not getting provisioned, follow the following steps:
+
+1. Get the events for the root resource using `kubectl describe` or `kubectl get event`
+2. If there are errors in the events, address them.
+3. If no errors, follow its sub-resources. `kubectl get <KIND> <NAME> -o=jsonpath='{.spec.resourceRef}{" "}{.spec.resourceRefs}' | jq`
+4. Go back to step 1 using one of resources returned by step 3.
+
+The rest of this section will show you how to debug issues related to compositions without using external tooling. However, if you are using ArgoCD or FluxCD with UI, you can visualize object relationships more easily in the UI. You can also use the kube-lineage plugin to visualize object relationships in your terminal.
```

Comment:
```suggestion
{{< hint "note" >}}
The rest of this section will show you how to debug issues related to compositions without using external tooling. 

If you are using ArgoCD or FluxCD with UI, you can visualize object relationships in the UI. 

You can also use the kube-lineage plugin to visualize object relationships in your terminal.
{{< /hint >}}
```

Wrapping in a hint since it's information not directly relevant to the main content.